diff --git a/megatron/core/transformer/attention.py b/megatron/core/transformer/attention.py
index 2200b5582..992fb650c 100644
--- a/megatron/core/transformer/attention.py
+++ b/megatron/core/transformer/attention.py
@@ -59,8 +59,8 @@ except ImportError:
     rearrange = None
 
 try:
-    from flash_attn_3.flash_attn_interface import _flash_attn_forward
-    from flash_attn_3.flash_attn_interface import (
+    from flash_attn_interface import _flash_attn_forward
+    from flash_attn_interface import (
         flash_attn_with_kvcache as flash_attn3_with_kvcache,
     )
 
